awx/main/scheduler/task_manager_models.py

```diff
@@ -18,10 +18,20 @@ class TaskManagerInstance:
     def __init__(self, obj):
         self.obj = obj
         self.node_type = obj.node_type
-        self.remaining_capacity = obj.capacity
+        self.consumed_capacity = 0
         self.capacity = obj.capacity
         self.hostname = obj.hostname
 
+    def consume_capacity(self, impact):
+        self.consumed_capacity += impact
+
+    @property
+    def remaining_capacity(self):
+        remaining = self.capacity - self.consumed_capacity
+        if remaining < 0:
+            return 0
```

Comment:
Do we not want to return a negative number here to show that the instance is over capacity?

Reply:
This is shared between the task manager and the API. From the API's perspective, `remaining_capacity` should probably get set to 0 when over-capacity for ease of understanding. From the perspective of the task manager's logic, the two "selection" methods have to work correctly, and I'm not worried about that, it handles 0 remaining capacity fine, I don't think there's any case where it would send work to a node that has 0 remaining capacity.